### PR TITLE
Added e2e tests for `post.published.edited` webhook

### PR DESCRIPTION
**Pull request by kritikash18**
Added e2e tests for post.published.edited webhook

Refs https://github.com/TryGhost/Ghost/issues/15537

### CODE REVIEW

**Comment by naz** [comment]
Hey-hey @kritikash18 👋  Could this possibly be a problem with matching the "author" propery like the discussed [here](https://github.com/TryGhost/Ghost/pull/15613#issuecomment-1280213239)? 🤔

**Comment by halldorbjarni** [comment]
Hey @kritikash18, I was curious if this was the same issue that I ran into. It turns out the routing issue is present here but it's actually the reason your single test suite run works. 

`yarn test:single test/e2e-webhooks/posts.test.js` generates a `post['current']['url'] = 'http://127.0.0.1:2369/404/'` due to the routing bug mentioned above and sends it to the webhook. Luckily this still matches the regex for `anyLocalURL` perfectly. 

However, `yarn test:all` generates the correct `post['current']['url'] = http://127.0.0.1:2369/testing-post-published-edited-webhook/'` **but** here the regex `anyLocalURL` doesn't match since it doesn't support dashes. To add it you can edit the `anyLocalUrl` matcher at `Ghost/ghost/core/test/utils/e2e-framework.js` to be something like `stringMatching(/http:\/\/127.0.0.1:2369\/\[w-]+\//),`. Other url allowed symbols could be added as well for completeness.

What I'm mostly curious about is why this matching error is not singled out in the output and instead the jest diff points at **all** the matchers as a potential problem. Any idea here @naz? I'm not sure if it's a jest bug or a bug in how Ghost directly invokes some internal jest packages.

**Comment by naz** [comment]
Oh hey @halldorbjarni 👋  Thanks for joining the discussion! 

>matching error is not singled out in the output

I've came across this issue few times now and I think it's worth the investigation and a separate issue in the test framework. It's super hard to figure out which parts did not match with the current output 💯

**Comment by halldorbjarni** [comment]
> I've came across this issue few times now and I think it's worth the investigation and a separate issue in the test framework. 

@naz I agree. It reproduces on a minimal example.  I've opened a [bug](https://github.com/facebook/jest/issues/13480) on jest. Let's get this fixed!

**Comment by kritikash18** [comment]
Updated the local URL to support - and it worked. Thanks for the pointers @naz and @halldorbjarni !

File: ghost/core/test/utils/e2e-framework.js [hunk 1]
```diff
@@ -384,7 +384,7 @@ module.exports = {
         // @NOTE: hack here! it's due to https://github.com/TryGhost/Toolbox/issues/341
         //        this matcher should be removed once the issue is solved - routing is redesigned
         //        An ideal solution would be removal of this matcher altogether.
-        anyLocalURL: stringMatching(/http:\/\/127.0.0.1:2369\/\w+\//),
+        anyLocalURL: stringMatching(/http:\/\/127.0.0.1:2369\/[A-Za-z0-9_-]+\//),
```
Comment (naz) [inline ghost/core/test/utils/e2e-framework.js:387]: ohhh! nice catch here 🎣

**Review by naz — APPROVED** [review]
Good to land in the main 👍  Thanks for great work @kritikash18!

**Comment by ErisDS** [comment]
@kritikash18 merged and labelled 🎉  Thanks so much for your continued contributions, please keep them coming 🙏

**Pull request merged**